util/webhook/webhook.go

```diff
@@ -258,6 +249,27 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 	}
 }
 
+// getWebUrlRegex compiles a regex that will match any targetRevision referring to the same repo as the given webURL.
+// webURL is expected to be a URL from an SCM webhook payload pointing to the web page for the repo.
+func getWebUrlRegex(webURL string) *regexp.Regexp {
+	urlObj, err := url.Parse(webURL)
+	if err != nil {
+		log.Warnf("Failed to parse repoURL '%s'", webURL)
+		return nil
+	}
```

Comment:
This is just a nit and not a request for change, but I was wondering whether it might be slightly better to change method signature to return `(*regexp.Regexp, error)` and then, instead of emitting a log message here (and the case below), do a `return nil, fmt.Errorf("Failed to...")` and let logging be done in the caller. This would probably save some logging during unit testing.

Reply:
Yeah I like that a lot better. Utility function should be a utility.